Assets/SEE/UI/Window/TreeWindow/DesktopTreeWindow.cs

```diff
@@ -405,15 +405,21 @@ void RegisterClickHandler()
 
                             // We want all applicable actions for the element, except ones where the element
                             // is shown in the TreeWindow, since we are already in the TreeWindow.
-                            IEnumerable<PopupMenuAction> actions = ContextMenuAction
-                                                                   .GetApplicableOptions(representedGraphElement,
-                                                                                         representedGameObject)
-                                                                   .Where(x => !x.Name.Contains("TreeWindow"));
-                            actions = actions.Append(new PopupMenuAction("Hide in TreeWindow", () =>
+                            List<PopupMenuAction> appends = new ()
                             {
-                                searcher.Filter.ExcludeElements.Add(representedGraphElement);
-                                Rebuild();
-                            }, Icons.Hide));
+                                new ("Hide in TreeWindow", () =>
+                                    {
+                                        searcher.Filter.ExcludeElements.Add(representedGraphElement);
+                                        Rebuild();
+                                    }, Icons.Hide)
+                            };
+                            IEnumerable<PopupMenuEntry> actions = ContextMenuAction
+                                                                   .GetOptionsForTreeView(contextMenu.ContextMenu,
+                                                                                         e.position,
+                                                                                         representedGraphElement,
+                                                                                         representedGameObject,
+                                                                                         appends);
+                            actions = actions.Concat(appends);
```

Comment:
Wouldn't it make more sense to define the `appends` after the `actions`? I.e., we could just directly do `actions = actions.Append(new(...))`. That way, the code is isomorphic to the menu.

Reply:
The appended actions must be passed to the context menu due to the submenus of the context menu. If the actions were bound as you suggest, they would disappear after the first time a submenu is opened and then returning to the main menu.